support snpEff names that contain multiple terms separated by "+"

bpow opened the pull request:
This may not be the perfect way to address #415 (the better way is #442), but it can be considered food for thought...

I was frustrated with having to use snpeff 3.6 (a couple of years old now!) to keep from missing splice site mutations, so I coded this up to split the impact_string by its `+` delimiters, sort the effects by lowest priority_code, and take the code of the 0th value. `consequence` and `so` still store `+` delimited versions, so the info from snpEff is not lost.

This may tide things over until settling in to ANN as per #442 (which should theoretically result in code reduction since VEP, snpEff, and Annovar all have plans to support ANN (hopefully in a not-too-incompatible way)

arq5x commented:
Thanks for this addition @bpow . Do you happen to have a small VCF that we could use for functional tests?

bpow commented:
In the process of thinking about coming up with a small file to test this, I noticed that `test/test.eff.vcf`contains a variant of the type that would be affected by this change, so I modified `test/test-eff.sh` to reflect this.

Specifically, `test.eff.vcf`contains a single variant, with effect `SPLICE_SITE_REGION+SYNONYMOUS_CODING`. Without the change proposed in this pull request, this effect string would not be recognized, so the `impact_severity` is set to `none`. With the proposed change, the `impact_severity` is `MED` (the higher of the two component parts: `SPLICE_SITE_REGION` has `MED` severity).

So, I added a check for the impact_severity in `test/test-eff.sh`. Let me know if there are other aspects of this pull request that you think should be tested, and I will try to come up with a vcf to use in testing.

I rebased and squashed in the changes to testing-related files.

brentp commented:
This is fixed in the recent PR that updates the impacts parsing.
